cpp/pixels-cli/main.cpp

```diff
@@ -25,141 +25,182 @@
 #include <iostream>
 #include <sstream>
 #include <string>
-#include <boost/algorithm/string.hpp>
-#include <boost/program_options.hpp>
+#include <vector>
+#include <map>
+#include <algorithm>
+#include <cctype>
 #include <executor/LoadExecutor.h>
+#include <iterator> // Add this line to include std::istream_iterator
 
-namespace bpo = boost::program_options;
+// Trim leading and trailing whitespace from a string
+std::string trim(const std::string& str)
+{
+  const auto first = str.find_first_not_of(' ');
+  if (first == std::string::npos)
+  {
+    return "";
+  }
+  const auto last = str.find_last_not_of(' ');
+  return str.substr(first, last - first + 1);
+}
+
+// Convert a string to lowercase
+std::string toLower(const std::string& str)
+{
+  std::string lowerStr = str;
+  std::transform(lowerStr.begin(), lowerStr.end(), lowerStr.begin(), [](unsigned char c) {
+    return std::tolower(c);
+  });
+  return lowerStr;
+}
+
+// Parse command-line arguments into a map
+std::map<std::string, std::string> parseArguments(const std::vector<std::string>& args)
```

Comment:
It is risky to replace boost parms with your own implementation. You'd better add the boost implementation back or make sure your implementation is robust enough.